src/t_set.c

```diff
@@ -1120,6 +1129,8 @@ void sunionDiffGenericCommand(client *c, robj **setkeys, int setnum,
             }
             setTypeReleaseIterator(si);
         }
+    } else if (op == SET_OP_DIFF && sameset) {
+        /* nothing to do... */
```

Comment:
if we put this one first (before the other `op == SET_OP_DIFF` ones, then there's no need to modify the other conditions.
alternatively if we do modify the other conditions, then there's no need for this new `else if`.
i prefer to put it first, and not modify the other conditions.

Reply:
Do you mean before this `if (op == SET_OP_DIFF && sets[0] && !sameset)`? This is not in the if-else chain. 
Or do you actually mean the first if  `if (op == SET_OP_UNION)` in the if-else chain?

Reply:
ohh, sorry.. i was looking at it through the narrow view of the github diff view.

Reply:
Never mind, so do we still need to change it?